prost-build/src/code_generator.rs

```diff
@@ -1036,6 +1016,38 @@ impl<'a> CodeGenerator<'a> {
         }
     }
 
+    /// Returns whether the Rust type for this field needs to be `Box<_>`.
+    ///
+    /// This can be explicitly configured with `Config::boxed`, or necessary
+    /// to prevent an infinitely sized type definition in case when the type of
+    /// a non-repeated message field transitively contains the message itself.
+    fn boxed(
+        &self,
+        field: &FieldDescriptorProto,
+        fq_message_name: &str,
+        oneof: Option<&str>,
+    ) -> bool {
+        if field.label == Some(Label::Repeated as i32) {
+            return false;
+        }
```

Comment:
Note that this changes the logic from the original: now any `boxed` match from the config will be ignored if the field is repeated, while previously it gave true. But I think that was wrong, because it never makes sense to box a `Vec`.

Reply:
To think of it, this should be reverted if we want this to merged for 0.12.x releases.

Reply:
The quick fix is to split this function into a `boxed_field` and `boxed_oneof` variants. Make sure to document why two variants are needed.

Maybe you could add a deprecation warning like this:
```rust
println!("cargo:warning=Field X is repeated and manually marked as boxed. This is deprecated and support will be removed in a later release");
```

Reply:
> The quick fix is to split this function into a `boxed_field` and `boxed_oneof` variants. Make sure to document why two variants are needed.

I don't like this idea. The logic would need to be replicated, even though it's mostly the same for regular fields and oneof members. More so with the deprecation warning, which I have now added as per your suggestion.